### DISCUSSION TITLE
how is this trained

### DISCUSSION THREAD

**Discussion by Wemmons831** [lucidrains/big-sleep#102] [Q&A]
Hello, I am a big fan of this project and use it quite often and I was just wondering how is this trained. My only idea of how it would be trained is to google search the words and then somehow combine all of them but I don't know if this is correct. So if anyone knows be educate me

**Comment by wolfgangmeyers** [comment]
The full details of how everything works are over my head I'm afraid, but I can explain at a high level:

Big Sleep is built on two lower-level components, BigGAN and CLIP.

Looking at [this article](https://machinelearningmastery.com/a-gentle-introduction-to-the-biggan/#:~:text=The%20BigGAN%20is%20an%20approach,(high%2Dfidelity)%20images.) I think BigGAN may have been trained on ImageNet data, pairing up images with classifiers.

CLIP is a pretrained model that was published by OpenAI, and there is an overview of how it was trained [here](https://openai.com/blog/clip/)
